internal/plugintest/config_test.go

```diff
@@ -0,0 +1,44 @@
+package plugintest
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"sync"
+	"sync/atomic"
+	"testing"
+)
+
+func TestDiscoverConfigConcurrently(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
```

Comment:
Tests requiring a temporary directory should prefer [`(testing.T).TempDir()`](https://pkg.go.dev/testing#T.TempDir)

```suggestion
	dir := t.TempDir()
```

Reply:
I keep forgetting about that API

Reply:
I'm probably going to just write the go/analysis thing for it and submit it for inclusion in golangci-lint, since it crops up in various places. We're [not alone](https://github.com/hashicorp/terraform/pull/30803). 😄